lib/task/jobs/jobWorkerTask.class.php

```diff
@@ -128,6 +140,17 @@ function () use (&$counter) {
                 }
             }
         );
+
+        $jobsCompleted = $worker->getJobsCompleted();
+
+        // Force worker's __destruct() to run.
+        $worker = null;
+
+        if ($this->maxJobCount > 0 && $this->maxJobCount <= $jobsCompleted) {
```

Comment:
This duplicates the logic in `AtomNetGearmanWorker::maxJobCountReached()`.  I think it would be better to use `if ($worker->maxJobCountReached()) {` and move the lines 146-147 that unset `$worker` to the end of this method, after this conditional.